### PR TITLE
[Feature] Deploy command

### PR DESCRIPTION
**Pull request by ilitteri**
## Motivation

This is a first iteration of the `aleo deploy` command that would let Aleo users deploy their programs to the `testnet3`. This is a WIP PR and it is not intended to be the final code (yet) since that would imply changes to snarkVM in order to follow the pre stablished code conventions.

## TODOs

Make the proper changes to snarkVM following the actual code convention.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -64,7 +64,7 @@ version = "1"
 [dependencies.snarkvm]
 #path = "../snarkVM"
 git = "https://github.com/AleoHQ/snarkVM.git"
-rev = "f7f5b13"
+branch = "feature/aleo-deploy"
```
Comment (raychu86) [inline Cargo.toml]: Stick with `rev`. There is another instance in `./rust`

File: cli/commands/deploy.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+pub struct Deploy;
+
+impl Deploy {
+    /// Deploys an Aleo program with the specified name.
+    pub fn parse(self) -> Result<String> {
+        // Derive the program directory path.
+        let path = std::env::current_dir()?;
+
+        // Load the package.
+        let package = Package::<Testnet3>::open(&path)?;
```
Comment (raychu86) [inline cli/commands/deploy.rs]: Please use `crate::Network`

**Comment by yhio** [comment]
can not deploy imported program
```sh
  external_call git:(feature/aleo_deploy) aleo deploy

⏳ Deploying 'external_call.aleo'...

 • Loaded universal setup (in 1973 ms)
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Cannot add program 'external_call.aleo' because its import 'difference.aleo' must be added first', /Users/me/.cargo/git/checkouts/snarkvm-f1160780ffe17de8/1db1f75/vm/package/deploy.rs:147:63
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by ilitteri** [comment]
> can not deploy imported program
> 
> ```shell
>   external_call git:(feature/aleo_deploy) aleo deploy
> 
> ⏳ Deploying 'external_call.aleo'...
> 
>  • Loaded universal setup (in 1973 ms)
> thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Cannot add program 'external_call.aleo' because its import 'difference.aleo' must be added first', /Users/me/.cargo/git/checkouts/snarkvm-f1160780ffe17de8/1db1f75/vm/package/deploy.rs:147:63
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> ```

Thanks for reporting this! We're working on the solution. You can follow it here if you want https://github.com/AleoHQ/snarkVM/pull/1004

File: cli/commands/deploy.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    pub fn parse(self) -> Result<String> {
+        // Derive the program directory path.
+        let path = std::env::current_dir()?;
+
+        // Load the package.
+        let package = Package::<Network>::open(&path)?;
+        println!("⏳ Deploying '{}'...\n", package.program_id().to_string().bold());
+
+        package.deploy::<Aleo>(Some("https://www.aleo.network/testnet3/deploy".to_string()))?;
+        println!();
```
Comment (raychu86) [inline cli/commands/deploy.rs:39]: What is this `println` for?

**Review by howardwu — APPROVED** [review]

**Comment by ilitteri** [comment]
> can not deploy imported program
> 
> ```sh
> 
>   external_call git:(feature/aleo_deploy) aleo deploy
> 
> 
> 
> ⏳ Deploying 'external_call.aleo'...
> 
> 
> 
>  • Loaded universal setup (in 1973 ms)
> 
> thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Cannot add program 'external_call.aleo' because its import 'difference.aleo' must be added first', /Users/me/.cargo/git/checkouts/snarkvm-f1160780ffe17de8/1db1f75/vm/package/deploy.rs:147:63
> 
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> 
> ```

You first need to deploy the imported program before deploying the program that imports it.

**Pull request merged**